src/codec/xz/encoder.rs

```diff
@@ -13,6 +13,13 @@ impl XzEncoder {
             inner: crate::codec::Xz2Encoder::new(crate::codec::Xz2FileFormat::Xz, level),
         }
     }
+
+    #[cfg(feature = "xz-parallel")]
+    pub fn parallel(level: u32, threads: u32) -> Self {
```

Comment:
```suggestion
    pub fn parallel(threads: NonZeroU32, level: u32) -> Self {
```

Let's make sure they have the same order of parameters and also use NonZeroU32